src/HospitalLibrary/Rooms/Repositories/RoomRepository.cs

```diff
@@ -25,11 +25,15 @@ public async Task<IEnumerable<Room>> FindAllByFloor(int floor)
         public async Task<IEnumerable<Room>> SearchByTypeAndName(RoomSearchDTO roomsSearchDTO, int floorId)
         {
             return await _dataContext.Rooms.Where(r =>
-                ((r.FloorId == floorId))).Where(r=>
-                 (r.RoomNumber.Contains(roomsSearchDTO.RoomName)&&
-                 ((r.RoomType == roomsSearchDTO.RoomType)||(roomsSearchDTO.RoomType==RoomType.NO_TYPE)))
+                ((r.FloorId == floorId))).Where(r =>
```

Comment:
You don't need parenthesis here, Rooms.Where(r => r.FloorId == floorId) is enough